agent/exec/container/executor.go

```diff
@@ -41,12 +41,35 @@ func (e *executor) Describe(ctx context.Context) (*api.NodeDescription, error) {
 		}
 	}
 
+	// add v1 plugins to 'plugins'
 	addPlugins("Volume", info.Plugins.Volume)
 	// Add builtin driver "overlay" (the only builtin multi-host driver) to
 	// the plugin list by default.
 	addPlugins("Network", append([]string{"overlay"}, info.Plugins.Network...))
 	addPlugins("Authorization", info.Plugins.Authorization)
 
+	// retrieve v2 plugins
+	v2plugins, err := e.client.PluginList(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// add v2 plugins to 'plugins'
+	for _, plgn := range v2plugins {
+		for _, typ := range plgn.Config.Interface.Types {
+			plgnTyp := typ.Capability
+			if typ.Capability == "volumedriver" {
+				plgnTyp = "Volume"
```

Comment:
@nishanttotla what happens if we remove these ifs ? The capability is `volumedriver`. I actually wonder if it shouldn't even be the whole type: `docker/volumedriver:1.0`. cc @anusha-ragunathan